### DISCUSSION TITLE
Disable automatic deployment completly, only allow manual deployment

### DISCUSSION THREAD

**Discussion by adarshaacharya** [vercel/community#5888] [Help] [answered]
### Summary

Hello team, 
Is there any way to ignore the automatic deployment completly and just create a manual deployment on vercel project.
I checked the ignore build step from Vercel deployment but seems like disabling it even cancels the `Create Deployment` i.e. manual deployment step.

### Steps to Reproduce

- Go to vercel settings and under Ignore Build Step section, select don't build anything.

![image](https://github.com/vercel/community/assets/50266088/af5823f8-4fb3-4865-81c1-0ec7f50fd1cc)

- Go to Deployment tab and select the Create Deployment, select branch and try to deploy the certain branch. eg : main

Now the build doesn't happens i.e. build will get cancelled before starting.

**Comment by amyegan** [comment, marked as answer]
Hey @adarshaacharya. You can set [`git.deploymentEnabled`](https://vercel.com/docs/projects/project-configuration/git-configuration#git.deploymentenabled) to false in a vercel.json file to disable automatic deployments

**Reply by adarshaacharya** [reply]
Thanks for reply @amyegan , but 

can we disable as per project? like i have one repo but have different projects in vercel  and want to have `projectA` to have automatic deployment enabled but want `projectB` `projectC` to have auto deployment disabled.

**Reply by adarshaacharya** [reply]
also using `**Create Deployment**`  feature doesn't deploy anything if we enable :
```
{
  "git": {
    "deploymentEnabled": {
      "main": false
    }
  }
}
```

in `vercel.json`..



Is it normal?

**Reply by adarshaacharya** [reply]
<img width="163" alt="image" src="https://github.com/vercel/community/assets/50266088/aa2349e8-967a-468e-a94f-5fefaee698fd">


I've to use `Promote this to production feature` to deploy anything to production.

**Reply by amyegan** [reply]
It sounds like you have some complexity in your repo setup and deployment workflow that I didn't know about. If you describe your process a bit more, we can work together to find a more appropriate solution. 🙏

**Reply by adarshaacharya** [reply]
@amyegan  So the first problem is that I want to disable auto deployment by default i.e. Don't want project to get deployed once I merge to main branch.

So as per the solution mentioned I created `vercel.json` on root of project (using nx as monorepo)

```
{
  "git": {
    "deploymentEnabled": {
      "main": false
    }
  }
}
```

 This doesn't build the project automatically but when I tried to do manual deployment by going doing `Create Deployment` in Deployments tab of vercel project, it doesn't build anything( i.e. manual deployment also doesn't work)

**Reply by adarshaacharya** [reply]
Hi @amyegan any solution for this

**Reply by swarnava** [reply]
>when I tried to do manual deployment by going doing Create Deployment in Deployments tab of vercel project, it doesn't build anything( i.e. manual deployment also doesn't work)

Can you record a screencast and share with us?

**Reply by adarshaacharya** [reply]
https://github.com/vercel/community/assets/50266088/3cfa6179-2502-43cd-bbe2-62df91c4214d

@swarnava Hre's the video and the repo link where I've setup `vercel.json` like this:

https://github.com/adarshaacharya/premier-league-points-table/blob/main/vercel.json

**Reply by swarnava** [reply]
Can you check Network request and see if there is any failed request?

**Reply by swarnava** [reply]
Nevermind, I was able to reproduce this and currently investigating.

**Reply by adarshaacharya** [reply]
Thanks @swarnava

**Reply by swarnava** [reply]
This issue should be fixed now.

**Reply by adarshaacharya** [reply]
That was quick. Thanks @swarnava
